### ISSUE TITLE
Multiline lists leave much to be desired 

### ISSUE THREAD

**Issue by Teslafly** [DiffSK/configobj#104]
Currently in configobj, multiline lists are implemented like so

```
[default]

list = '''
    item1,
    item2, #indented for clarity
    item3,
    etc,
'''
```

If items in the list are indented, when parsed the whitespace is also included, along with single line comments:

```
>>> from configobj import ConfigObj as cfobj
>>> example = cfobj('example.conf')
>>> print(dict(example))
{'default': {'list': '\n    item1,\n    item2, #indented for clarity\n    item3,\n    etc,\n'}}
>>> list = example['default']['list'].splitlines()
>>> list
['', '    item1,', '    item2, #indented for clarity', '    item3,', '    etc,']
>>> list = [x.strip() for x in list if x]
>>> list
['item1,', 'item2, #indented for clarity', 'item3,', 'etc,']
```
As you can see, there is a large amount of additional work needed to parse this list into a usable state. not to mention that single line comments don't work and commas aren't parsed correctly.

Contrast this with how configparser implements lists.

```
[default]

list:
    item1
    item2 #comments don't work
    item3
    etc
'''
```
and the code:
```
>>> import configparser
>>> example = configparser.SafeConfigParser()
>>> example.read('example2.conf')
['example2.conf']
>>> print(example['default']['list'])

item1
item2 #indented for clarity
item3
etc
>>> list = example['default']['list'].splitlines()
>>> print(list)
['', 'item1', 'item2 #indented for clarity', 'item3', 'etc']
>>> list = [x for x in stuff if x]
>>> print(list)
['item1', 'item2 #indented for clarity', 'item3', 'etc']
```

configparser also doesn't like comments in this case, but it still requires less work with less chance for error to parse it. This is beacuse it removes the beforehand whitespace and doesn't need a blockquote to define the value. merging the advantages of onfigobj and configparser, I propose the following behavior.

```
[default]

list:
    item1,
    item2, #indented for clarity
    item3,
    etc,
```

```
>>> from configobj import ConfigObj as cfobj
>>> example = cfobj('example.conf')
>>> print(dict(example))
{'default': {'list': ['item1', 'item2', 'item3',  'etc']}}
>>> list = example['default']['list']
>>> list
['item1', 'item2', 'item3',  'etc']
```

This preserves the congigobj list syntax, while allowing multiline expanded lists.
it does this by using the ':' to expect an indented list, removes the indent, and initially splits the list by newlines. Then it removes comments after the ',' seperator to allow for per-line comments. (before and it may be part of the data) 

What do you think about this behavior? Does it remain consistent with the configobj objectives?

This is probably an enhancement.

**Comment by EliAndrewC** [comment]
This seems like a good proposal, though I wouldn't want this to happen automatically.  In your example, you didn't have a specfile saying that ``list`` was of type ``string_list``, nor did you use ``.as_list()``.  I wouldn't want to automatically convert string values to lists just because they fell across multiple lines and had a comma.

A modified version of your proposal would be that when we have a specfile such as

```
fruits = string_list(default=list())
```

And a config file that looks like

```
fruits:
    apples,
    bannanas,  # good source of potassium
    cherries
```

Then you could say

```python
spec = ConfigObj('configspec.ini', list_values=False, _inspec=True)
conf = ConfigObj('example.conf', configspec=spec)
assert conf.validate(Validator())
assert conf['fruits'] == ['apples', 'bannanas', 'cherries']
```

Or in cases where there is no specfile, we could just say

```
conf = ConfigObj('example.conf')
assert conf.as_list('fruits') == ['apples', 'bannanas', 'cherries']
```

I don't forsee any backwards-compatibility issues from that, since in both cases we're making something work which would not have correctly parsed in previous versions, so there's no chance of breaking anyone's existing code.

I'm dubious about when I'll next have time to do this kind of work on ConfigObj, but a pull request would be welcome.  (My job keeps promising that they'll be able to spare me for a little while to do this kind of open source work, but that keeps getting postponed.)

**Comment by jhermann** [comment]
If you can't find / get the time, maybe adding some people to the project might help. For example, the 5.1.0 milestone is pretty much done, it just has to be pushed out…

**Comment by EliAndrewC** [comment]
Yeah, I feel bad about letting that slip for so many months.  Are you offering to help out?  I'll happily add you as one of the project maintainers if you've got the time and inclination, given how active you've been on this project.

**Comment by Teslafly** [comment]
I may be able to implement this feature, I am asking for it after all. Though as a mediocre python programmer I can't promise that it will be the most efficient or clean.

**Comment by jhermann** [comment]
@EliAndrewC Yeah, though I'm also spread thin across too many projects, I am in the comfortable position to be able to spend some company time on OSS projects. If we spread the load a little more, the frequency of releases should rise.

**Comment by EliAndrewC** [comment]
Thanks so much - I'll be grateful for whatever you're able to offer.  (My company theoretically lets us spend time on OSS, but in practice we've been in crunch mode for awhile and I'm not sure if/when that'll change.)

I've just added you as a collaborator on this project in Github, but I'll also need to add you to PyPI.  Do you have an account there, and if so then what is your username?

**Comment by jhermann** [comment]
@EliAndrewC It's simply "jh". Also Rob needs to add me (and you?) to "validate", see #105.

**Comment by robdennis** [comment]
I will add jh and Eli's name to validate

**Comment by robdennis** [comment]
I've added "jh" as maintainer to validate/configojb and EliAndrewC as owner to validate

**Comment by eight04** [comment]
This bug makes configobj unable to read config files generated by configparser. Any suggestion?

**Comment by nelsonag** [comment]
Hi all, I'm curious on the status of this issue. We are using ConfigObj for production work and am at the point where I need to decide if I should implement something locally (quick and dirty) to deal with multi-line lists in the input or if this issue is expected to be resolved and merged soon. 

Thanks!
